aten/src/ATen/native/cuda/ReplicationPadding.cu

```diff
@@ -0,0 +1,239 @@
+#include "ATen/ATen.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include "ATen/cuda/CUDAContext.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/TensorUtils.h"
+#include "ATen/Utils.h"
+#include "c10/util/Exception.h"
+#include "THC/THCGeneral.h"
+#include "THC/THCNumerics.cuh"
+#include "THC/THCDeviceTensor.cuh"
+#include "THC/THCDeviceUtils.cuh"
+
+#include <algorithm>
+#include <cfloat>
+#include <cmath>
+
+
+namespace at {
+namespace native {
+__host__ __device__ __forceinline__ int imin(int a, int b) {
+      return a > b ? b : a;
+}
+
+__host__ __device__ __forceinline__ int imax(int a, int b) {
+      return a > b ? a : b;
+}
+
+__host__ __device__ __forceinline__ int iabs(int a) {
+      return a >= 0 ? a : -a;
+}
+
+namespace {
+template <typename Dtype>
+  __global__ void replication_pad(
+      const PackedTensorAccessor<Dtype, 3>& input,
+      PackedTensorAccessor<Dtype, 3>& output,
```

Comment:
I don't think we want to pass the `PackedTensorAccessor`s by reference. To me, the main thing about them (compared to `TensorAccessor` is that they carry their sizes/strides).
I think DeviceTensor worked the same way.

Reply:
Oooh, that's exactly right. I feel embarrassed for missing it!

Reply:
You found so many bugs in my PRs, I'm glad that after so many of them, I get to return the favour. (And I only saw it after debugging in the direction of *are the offsets into the tensors invalid?* for quite a while...)